choreography/src/choreography/Fabrikam.Choreography.ChoreographyService/Controllers/ChoreographyController.cs

```diff
@@ -56,12 +56,15 @@ public async Task<IActionResult> Operation([FromBody] EventGridEvent[] events)
                 return BadRequest("No Event for Choreography");
             }
                 
-            if (events[0].EventType is EventTypes.EventGridSubscriptionValidationEvent)
+            if (events[0].EventType is "Microsoft.EventGrid.SubscriptionValidationEvent")
             {
                 try
                 {
                     var data = Operations.ConvertDataEventToType<SubscriptionValidationEventData>(events[0].Data);
-                    var response = new SubscriptionValidationResponse(data.ValidationCode);
+                    object[] parameters = new object[1];
+                    parameters[0] = data.ValidationCode;
+                    object obj = Activator.CreateInstance(typeof(SubscriptionValidationResponse), parameters);
+                    SubscriptionValidationResponse response = (SubscriptionValidationResponse) obj;
```

Comment:
This looks odd.

Reply:
Because there is no constructor in class `SubscriptionValidationResponse`, it can only be instantiated by reflection.

Reply:
That is definitely not something that we want to do in a sample.   If the type is not offering a constructor, then that's a strong signal that it is not intended to be created from developer code outside of unit test scenarios.

In this case, it seems like we should be using [`TryGetSystemEventData`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventgrid/Azure.Messaging.EventGrid/samples/Sample3_ParseAndDeserializeEvents.md#using-trygetsystemeventdata) to retrieve the strongly-typed event and sending back a custom response based on the `ValidationCode` member.

Reply:
Thanks, I updated this by using `EventGridModelFactory ` according to your other comment.